tests/phpunit/tests/test-switch-blog-custom-code.php

```diff
@@ -0,0 +1,145 @@
+					$post->ID,
+					array(
+						$languages[0]['slug'] => $post->ID,
+						$languages[1]['slug'] => $tr_post->ID,
+					)
+				);
+			}
+		}
+
+		public function test_switch_blog_with_custom_code() {
+			$pll_frontend = $this->get_pll_frontend_env();
+			$pll_frontend->curlang = $pll_frontend->model->get_language( 'en' ); // English is a common language for blogs with Polylang activated, @see {PLL_Multisites_TestCase::set_up()}.
+			do_action( 'pll_language_defined' );
+			do_action_ref_array( 'pll_init', array( &$pll_frontend ) );
+			$GLOBALS['polylang'] = $pll_frontend;
+
+			/*
+			 * Test blog with Polylang activated and pretty permalink with language as directory.
+			 */
+			switch_to_blog( (int) $this->blog_with_pll_directory->blog_id );
+
+			$home = get_home_url();
+
+			$this->assertNotEmpty( $home );
+
+			$this->assertInstanceOf( PLL_Language::class, $pll_frontend->curlang );
+
+			$post_id = pll_get_post( $this->posts[ (int) $this->blog_with_pll_directory->blog_id ]->ID );
+
+			$this->assertNotFalse( $post_id );
+			$this->assertSame( $this->posts[ (int) $this->blog_with_pll_directory->blog_id ]->ID, $post_id );
+
+			restore_current_blog();
+
+			/*
+			 * Test blog with Polylang activated and pretty permalink with language as domains.
+			 */
+			switch_to_blog( (int) $this->blog_with_pll_domains->blog_id );
+
+			$home = get_home_url();
+
+			$this->assertNotEmpty( $home );
+
+			$this->assertInstanceOf( PLL_Language::class, $pll_frontend->curlang );
+
+			$post_id = pll_get_post( $this->posts[ (int) $this->blog_with_pll_domains->blog_id ]->ID );
+
+			$this->assertNotFalse( $post_id );
+			$this->assertSame( $this->posts[ (int) $this->blog_with_pll_domains->blog_id ]->ID, $post_id );
+
+			restore_current_blog();
+
+			/*
+			 * Test blog with Polylang activated and plain permalink.
+			 */
+			switch_to_blog( (int) $this->blog_with_pll_plain_links->blog_id );
+
+			$home = get_home_url();
+
+			$this->assertNotEmpty( $home );
+
+			$this->assertInstanceOf( PLL_Language::class, $pll_frontend->curlang );
+
+			$post_id = pll_get_post( $this->posts[ (int) $this->blog_with_pll_plain_links->blog_id ]->ID );
+
+			$this->assertNotFalse( $post_id );
+			$this->assertSame( $this->posts[ (int) $this->blog_with_pll_plain_links->blog_id ]->ID, $post_id );
+
+			restore_current_blog();
+
+			/*
+			 * Test blog with Polylang deactivated and pretty permalink.
+			 */
+			switch_to_blog( (int) $this->blog_without_pll_pretty_links->blog_id );
+
+			$home = get_home_url();
+
+			$this->assertNotEmpty( $home );
+
+			// Even though Polylang is deactivated on the current blog, the current language must be kept.
+			$this->assertInstanceOf( PLL_Language::class, $pll_frontend->curlang );
+
+			$post_id = pll_get_post( $this->posts[ (int) $this->blog_without_pll_pretty_links->blog_id ]->ID );
+
+			$this->assertNotFalse( $post_id );
+			// No language, no post.
+			$this->assertSame( 0, $post_id );
```

Comment:
`assertNotFalse()` isn't necessary then.

Reply:
Well I do want to test the result isn't `false`, because it doesn't mean the same as `0`.
I the result is `false`, it means the current language has been lost.

Reply:
You already test it's not `false` with `assertSame()` since it does a strict comparison. Or am I wrong?

Reply:
Well, I'm stupid...

Reply:
Removed in 6fc67034.

Reply:
![Homer Boo Boo](https://github.com/polylang/polylang/assets/1878479/3014eb9d-539f-4fea-aeea-754927dfe3b2)